app/helpers/data_getter.py

```diff
@@ -715,7 +715,13 @@ def get_locations_of_events():
             for event in DataGetter.get_live_and_public_events():
                 if not string_empty(event.location_name) and not string_empty(event.latitude) and not string_empty(
                      event.longitude):
-                    names.append(event.location_name.split(',')[0])
```

Comment:
> Instead of a request, try using searchable_location_name.

@niranjan94 I just checked. the problem with searchable_location_name is it was storing short_name instead of the long_name. And since it is already stored in the db, it is showing the wrong output. So I can improve it for the later events, but already stored locations would still be wrong.

> Request to the API is gonna be very slow without caching

I know it's gonna be slow. So I will try implementing the cache once more and check.